orchestra/utils/task_lifecycle.py

```diff
@@ -290,6 +303,14 @@ def reassign_assignment(worker_id, assignment_id):
     if assignment.task.is_worker_assigned(worker):
         raise TaskAssignmentError('Worker already assigned to this task.')
 
+    if assignment.status == TaskAssignment.Status.FAILED:
```

Comment:
This state is only for machine tasks actually, we need to refactor the restaffing logic to remove setting a human task to the `FAILED` state: https://github.com/unlimitedlabs/orchestra/blob/master/orchestra/utils/task_lifecycle.py#L158